frontend/javascripts/admin/user/user_list_view.tsx

```diff
@@ -162,51 +142,41 @@ class UserListView extends React.PureComponent<Props, State> {
     });
     Promise.all(newUserPromises).then(
       (newUsers) => {
-        this.setState({
-          users: newUsers,
-          selectedUserIds: [selectedUser.id],
-        });
+        setUsers(newUsers);
+        setSelectedUserIds([selectedUser.id]);
         Toast.success(messages["users.change_email_confirmation"]);
-        if (this.props.activeUser.email === selectedUser.email) Store.dispatch(logoutUserAction());
+        if (activeUser.email === selectedUser.email) Store.dispatch(logoutUserAction());
       },
       () => {}, // Do nothing, change did not succeed
     );
-  };
+  }
 
-  handleUsersChange = (updatedUsers: Array<APIUser>): void => {
-    this.setState({
-      users: updatedUsers,
-      isExperienceModalOpen: false,
-      isTeamRoleModalOpen: false,
-    });
-  };
+  function handleUsersChange(updatedUsers: Array<APIUser>): void {
+    setUsers(updatedUsers);
+    setIsExperienceModalOpen(false);
+    setIsTeamRoleModalOpen(false);
+  }
 
-  closeExperienceModal = (updatedUsers: Array<APIUser>): void => {
+  function closeExperienceModal(updatedUsers: Array<APIUser>): void {
     const updatedUsersMap = _.keyBy(updatedUsers, (u) => u.id);
 
-    this.setState((prevState) => ({
-      isExperienceModalOpen: false,
-      users: prevState.users.map((user) => updatedUsersMap[user.id] || user),
-      singleSelectedUser: null,
-      selectedUserIds: prevState.singleSelectedUser == null ? [] : prevState.selectedUserIds,
-    }));
-  };
+    setIsExperienceModalOpen(false);
+    setUsers(users.map((user) => updatedUsersMap[user.id] || user));
```

Comment:
Here is another case where in the previous version the state change depended on the previous state. Therefore, I suggest to pass a function to the set method kinda like:

```suggestion
    setUsers((users) => users.map((user) => updatedUsersMap[user.id] || user));
```